### PR TITLE
HorizontalPodAutoscaler status

### PR DESCRIPTION
**Pull request by mwielgus**
Adds Status information to HorizontalPodAutoscaler experimental API object.

### CODE REVIEW

**Comment by mwielgus** [comment]
cc: @jszczepkowski

**Comment by jszczepkowski** [comment]
CC @davidopp

File: pkg/expapi/types.go [hunk 1]
```diff
@@ -74,8 +75,8 @@ type SubresourceReference struct {
 	Subresource string `json:"subresource,omitempty"`
 }
 
-// TargetConsumption is an object for specifying target average resource consumption of a particular resource.
-type TargetConsumption struct {
+// Consumption is an object for specifying average resource consumption of a particular resource.
+type Consumption struct {
```
Comment (jszczepkowski) [inline pkg/expapi/types.go]: Maybe ResourceConsumption will be a better name? Consumption seems to be to broad for me.

**Comment by jszczepkowski** [comment]
CC @bgrant0607

File: pkg/expapi/types.go (hunk 1 above)
Comment (davidopp) [inline pkg/expapi/types.go]: +1

**Comment by jszczepkowski** [comment]
LGTM

File: pkg/expapi/types.go [hunk 2]
```diff
@@ -91,7 +92,27 @@ type HorizontalPodAutoscalerSpec struct {
 	MaxCount int `json:"maxCount"`
 	// Target is the target average consumption of the given resource that the autoscaler will try to maintain by adjusting the desired number of pods.
 	// Currently two types of resources are supported: "cpu" and "memory".
-	Target TargetConsumption `json:"target"`
+	Target ResourceConsumption `json:"target"`
```
Comment (bgrant0607) [inline pkg/expapi/types.go]: This is intended to be a single resource only? And is an absolute value that itself must be vertically auto-sized?

File: pkg/expapi/types.go [hunk 3]
```diff
@@ -91,7 +92,27 @@ type HorizontalPodAutoscalerSpec struct {
 	MaxCount int `json:"maxCount"`
```
Comment (bgrant0607) [inline pkg/expapi/types.go]: You don't plan to set either minCount or maxCount by default?

File: pkg/expapi/types.go (hunk 2 above)
Comment (davidopp) [inline pkg/expapi/types.go]: > This is intended to be a single resource only?

Yes, the design doc says "This is intended to be a single resource only" in the algorithm section.

> And is an absolute value that itself must be vertically auto-sized?

What does this mean? Are you asking what is the behavior when you do vertical and horizontal simultaneously?

File: pkg/expapi/types.go (hunk 3 above)
Comment (jszczepkowski) [inline pkg/expapi/types.go]: It is possible. We can set minCount to 1 by default and maxCount to Scale.Replicas. WDYT?

**Comment by jszczepkowski** [comment]
Part of #12087.

**Pull request merged**